src/components/Visualization.tsx

```diff
@@ -40,7 +38,10 @@ const Visualization = () => {
   const previousLayout = useRef(visualizationStore.layout);
   const previousIntegrationJson = useRef(integrationJsonStore.integrationJson);
   const visualizationService = new VisualizationService(integrationJsonStore, visualizationStore);
-  const stepsService = new StepsService(integrationJsonStore, nestedStepsStore, visualizationStore);
+  const stepsService = useMemo(
```

Comment:
:eyes: just curious, should `useMemo()` be the default way of instantiating (and caching) service instances? a "memo" for #1233

Reply:
That's a very interesting question :smiley: 

From my point of view, I don't think that this is the definite way to go but a transition point.

At first, the `onNodeClick` function was being recreated on every render pass, together with the `stepService`, so as I wrapped `onNodeClick` function into a `useCallback()` hook to avoid recreating it, `stepService` now it's a required dependency of it, so I wrapped the service with `useMemo()` to recreate it only if something change.

I think that a more appropriate fix would be to have this service [as a context](https://beta.reactjs.org/reference/react/useContext), this way we wouldn't need to create the service in different places but just call the `useContext()` hook where needed.

Reply:
aha! makes sense, thanks :+1: